response-types.js

```diff
@@ -1,27 +1,78 @@
-class Response {
-  constructor (data, meta = undefined) {
-    this.data = data
-    this.errors = []
-    if (meta) { this.meta = meta }
-  }
-}
+const statuses = require('statuses')
+const util = require('./utils/util')
+
+const codes = [200, 201, 202]
 
-class Error {
-  constructor (title) {
+class FSError extends Error {
+  constructor ({code, title, status = '500'}, extraProps = {}) {
+    super(title)
+    this.code = code
     this.title = title
+    this.status = status
+    Object.keys(extraProps).forEach(key => {
+      this[key] = extraProps[key]
+    })
+  }
+
+  static userDoesNotExist ({title = 'User doesn\'t exist', status = '404'} = {}) {
```

Comment:
You should not be allowed to change the title according to http://jsonapi.org/format/#error-objects. We should add an optional message though, allowing to be a bit more specific about the issue